hpe3parclient/client.py

```diff
@@ -5373,4 +5426,514 @@ def remove_vlun_nvme(self, vol_name_3par, hostname, host_nqn):
                              " belongs to different host: %(vlun_host)s",
                              {'lun': vlun['lun'],
                               'vlun_host': vlun.get('hostname', 'Unknown')})
+                
+
+    
+    def _check_license_enabled(self, systemVersion, valid_licenses,
+                               license_to_check, capability):
+        """Check a license against valid licenses on the array."""
+        major_minor = systemVersion.split('.')
+        if int(major_minor[0]) == 10 and int(major_minor[1]) >= 5:
+            return True
+        else:
+            if valid_licenses:
+                for license in valid_licenses:
+                    if license_to_check in license.get('name'):
+                        return True
+                    logger.debug("'%(capability)s' requires a '%(license)s' "
+                          "license which is not installed.",
+                            {'capability': capability,
+                            'license': license_to_check})
+                return False
+
+
+    def _get_3par_vol_comment_value(self, vol_comment, key):
+        comment_dict = dict(ast.literal_eval(vol_comment))
+        if key in comment_dict:
+            return comment_dict[key]
+        return None
+
+
+    def _get_key_value(self, hpe3par_keys, key, default=None):
```

Comment:
The variable "hpe3par_keys" is used by cinder. So this function should be part of common.py